### DISCUSSION TITLE
Which worker pool should be used?

### DISCUSSION THREAD

**Discussion by mschorsch** [quarkusio/quarkus#28862] [Q&A] [answered]
There are several worker pools in Quarkus.
- Vert.x worker pool via [executeBlocking](https://vertx.io/docs/apidocs/io/vertx/core/Context.html#executeBlocking-io.vertx.core.Handler-)
- Mutiny worker pool via [getDefaultExecutor](https://javadoc.io/static/io.smallrye.reactive/mutiny/1.7.0/io/smallrye/mutiny/infrastructure/Infrastructure.html#getDefaultExecutor())
- [Managed Executor](https://quarkus.io/guides/context-propagation#usage-example-for-completionstage)

Which pool should be used by default? How to configure these pools via `application.conf`?

**Comment by mschorsch** [comment]
@geoand Maybe you can answer this or ping the right person?

**Reply by geoand** [reply]
I'll answer next week when I'm back

**Comment by geoand** [comment, marked as answer]
If you are asking which worker pool application code should be using, then the answer is that the Managed Executor from Smallrye Context Propagation is almost certainly what you want.

**Reply by mschorsch** [reply]
Can I configure the pool via the application.conf (min and max size for example)? Currently the pool seems unrestricted. I didn't found anything in the documentation.

**Reply by geoand** [reply]
Indeed it doesn't look like it's configurable... That's unfortunate... Can you open an issue asking for such a feature?

**Reply by mschorsch** [reply]
https://github.com/quarkusio/quarkus/issues/28977
